### PR TITLE
Seperate SinkA/Prefetch to fix timing

### PR DESCRIPTION
**Pull request by yulightenyu**
a. do not check aMerge when prefetch
b. prefetch duplicate with MSHR check move from s1 to s2
c. when prefetch dup MSHR -> drop @S2

### CODE REVIEW

File: src/main/scala/coupledL2/RequestBuffer.scala [hunk 1]
```diff
@@ -90,10 +91,31 @@ class RequestBuffer(flow: Boolean = true, entries: Int = 4)(implicit p: Paramete
...
+  } else {
+    in := io.in.bits
+  }
+
   /* ======== Data Structure ======== */
 
-  io.ATag := io.in.bits.tag
-  io.ASet := io.in.bits.set
+//  io.ATag := io.in.bits.tag
+//  io.ASet := io.in.bits.set
```
Comment (linjuanZ) [inline src/main/scala/coupledL2/RequestBuffer.scala:116]: Remove stale comments

File: src/main/scala/coupledL2/RequestBuffer.scala [hunk 2]
```diff
@@ -131,19 +153,19 @@ class RequestBuffer(flow: Boolean = true, entries: Int = 4)(implicit p: Paramete
 //  }
 
   // other flags
-  val in      = io.in.bits
+//  val in      = io.in.bits
```
Comment (linjuanZ) [inline src/main/scala/coupledL2/RequestBuffer.scala:156]: Remove stale comments

File: src/main/scala/coupledL2/RequestBuffer.scala [hunk 3]
```diff
@@ -161,17 +183,17 @@ class RequestBuffer(flow: Boolean = true, entries: Int = 4)(implicit p: Paramete
...
+  io.hasMergeA := mergeA && inVal && !sameAddr(in, RegNext(in))
 
   //  val depMask    = buffer.map(e => e.valid && sameAddr(io.in.bits, e.task))
   // remove duplicate prefetch if same-addr A req in MSHR or ReqBuf
   val isPrefetch = in.fromA && in.opcode === Hint
   val dupMask    = VecInit(
-    io.mshrInfo.map(s =>
-      s.valid && s.bits.isAcqOrPrefetch && sameAddr(in, s.bits)) ++
+//    io.mshrInfo.map(s =>
+//      s.valid && s.bits.isAcqOrPrefetch && sameAddr(in, s.bits)) ++
```
Comment (linjuanZ) [inline src/main/scala/coupledL2/RequestBuffer.scala:196]: Remove stale comments

File: src/main/scala/coupledL2/SinkA.scala [hunk 4]
```diff
@@ -111,6 +112,17 @@ class SinkA(implicit p: Parameters) extends L2Module {
...
+    io.taskPrefetch.get.bits := fromPrefetchReqtoTaskBundle(io.prefetchReq.get.bits)
+    io.prefetchReq.get.ready := io.taskPrefetch.get.ready && !io.a.valid
+  }
+  
+
+  io.task.valid := io.a.valid
+  io.task.bits := fromTLAtoTaskBundle(io.a.bits)
+  io.a.ready := io.task.ready
+
+/*  if (prefetchOpt.nonEmpty) {
```
Comment (linjuanZ) [inline src/main/scala/coupledL2/SinkA.scala:125]: Remove stale comments

**Review by linjuanZ — CHANGES_REQUESTED** [review]

File: src/main/scala/coupledL2/RequestBuffer.scala (hunk 1 above)
Comment (yulightenyu) [reply src/main/scala/coupledL2/RequestBuffer.scala]: deleted

File: src/main/scala/coupledL2/RequestBuffer.scala (hunk 2 above)
Comment (yulightenyu) [reply src/main/scala/coupledL2/RequestBuffer.scala]: deleted

File: src/main/scala/coupledL2/RequestBuffer.scala (hunk 3 above)
Comment (yulightenyu) [reply src/main/scala/coupledL2/RequestBuffer.scala]: deleted

File: src/main/scala/coupledL2/SinkA.scala (hunk 4 above)
Comment (yulightenyu) [reply src/main/scala/coupledL2/SinkA.scala]: deleted

**Review by linjuanZ — APPROVED** [review]

**Pull request merged**